### ISSUE TITLE
almost identical `await statement outside of a function` and `await outside of an asynchronous function` errors

### ISSUE THREAD

**Issue by samuelcolvin** [astral-sh/ty#2598]
### Summary

Running `uvx ty example.py` on

```py
async def foo():
    return 123


await foo()
```

Emits:

```
error[invalid-syntax]: `await` statement outside of a function
 --> example.py:5:1
  |
5 | await foo()
  | ^^^^^^^^^^^
  |

error[invalid-syntax]: `await` outside of an asynchronous function
 --> example.py:5:1
  |
5 | await foo()
  | ^^^^^^^^^^^
  |

Found 2 diagnostics
```

### Version

ty 0.0.13 (fc1478bd9 2026-01-21)

**Comment by AlexWaygood** [comment]
Thanks! To clarify, the bug here is that two syntax errors are emitted instead of 1? That does seem suboptimal

**Comment by samuelcolvin** [comment]
sorry, yes exactly - there's one thing wrong here, I expect one error.

**Comment by AlexWaygood** [comment]
Cc. @ntBre

**Comment by MichaReiser** [comment]
The comment here seems relevant


https://github.com/astral-sh/ruff/blob/main/crates/ruff_python_parser/src/semantic_errors.rs#L982-L984

We should avoid emiting both syntax errors, given that ty doesn't require the same backwards compatibility
